### ISSUE TITLE
upcoming cards: 3-dots menu (Card details)  not working

### ISSUE THREAD

**Issue by cloud2018** [nextcloud/deck#2441]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Describe the bug**
3-dots menus of a card in "upcoming card" has unexpected results (Card detauils and Move card)

**To Reproduce**
1. Open upcoming cards
2. Click on 3 dots menu of a card
3. "Card details"
3.1. Click on "Card details"
3.1. See never ending circle and "Loading board"
or 
4. "Move card"
4.1  Click on "Move card"
4.2. Nothing happens

By the way:
Papercut: "Upcoming cards" doesn't show only upcoming cards, also overdue cards. Perhaps "My cards with due date" is more appropriate 

**Versions**
Deck 1.1.1
NC 20

**Comment by juliushaertl** [comment]
Any errors reported in the browser console?

**Comment by cloud2018** [comment]
browser console:
http://xxxxxxxxxxx/nextcloud/index.php/apps/deck/boards/NaN
TypeError: this.board is null

**Comment by juliushaertl** [comment]
Could you check if that is still the case on the latest release 1.1.2?

**Comment by DragonDev1906** [comment]
This could be related: When I click "Card details" the first time the url changes from "/apps/deck#/board/6" to "/apps/deck#/" but nothing happens (No console output either). The second time I click "Card details" nothing happens and I get the following error in the browser console:
```
Uncaught (in promise) NavigationDuplicated: Avoided redundant navigation to current location: "/".
    Me vue-router.esm.js:1995
    confirmTransition vue-router.esm.js:1965
    transitionTo vue-router.esm.js:2190
    push vue-router.esm.js:2634
    push vue-router.esm.js:2950
    push vue-router.esm.js:2949
    openCard CardMenu.vue:128
    VueJS 4
    onClick ncvuecomponents.js:66
    VueJS 49
    openMenu ncvuecomponents.js:151
    VueJS 4
    show v-tooltip.esm.js:3600
    r v-tooltip.esm.js:3846
    _addEventListeners v-tooltip.esm.js:3857
    $_addEventListeners v-tooltip.esm.js:3839
    $_init v-tooltip.esm.js:3641
    mounted v-tooltip.esm.js:3576
    VueJS 23
vue-router.esm.js:1995:14
    transitionTo vue-router.esm.js:2210
    s vue-router.esm.js:2247
    confirmTransition vue-router.esm.js:2258
    transitionTo vue-router.esm.js:2190
    push vue-router.esm.js:2634
    push vue-router.esm.js:2950
    push vue-router.esm.js:2949
    openCard CardMenu.vue:128
    openCard self-hosted:1161
    VueJS 4
    onClick ncvuecomponents.js:66
    onClick self-hosted:1161
    VueJS 33
```

I don't see a loading indicator but nothing happens.

Deck Version: 1.1.2

**Comment by juliushaertl** [comment]
OK, i can actually reproduce that:

![image](https://user-images.githubusercontent.com/3404133/96120777-5c184880-0eef-11eb-8f59-6f28808bda7b.png)

**Comment by ei8fdb** [comment]
All the behaviour @cloud2018 has mentioned above I have replicated, and I can add some more, If a card is assigned to a user, it is possible to assign/unassign it ![as can be seen in this short screenshot clip](https://user-images.githubusercontent.com/1899506/98098033-b456c080-1e85-11eb-8540-df3f82ac76aa.gif)

**Comment by matthijskooijman** [comment]
I was having the same problem (NC 21.0.1, Deck 1.3.something) every single time I tried to open the board details, making them completely inacessible. Happened both in Firefox and Chromium. I'm pretty sure it used to work a few weeks ago, when I set up this board and shared it with people (though that was probably still with NC 20 and an older deck too).

Then I just upgraded to Deck 1.4.1, and now the problem seems to have gone away. Not sure if there is anything in that version that fixed it, or that the act of upgrading itself (maybe temporarily) fixed the problem, though..

**Comment by juliushaertl** [comment]
Yes, there was indeed some rework on the handling of how cards are displayed if it is not shown on a regular board view or if the cards are from a different board in #2934. The card should no longer depend on the current board for the individual actions since that so I consider this fixed.

**Comment by matthijskooijman** [comment]
Hm, I just noticed that I had completely misread this issue: The original issue is about *card* details, while my comment was about *board* details. Maybe the underlying cause was the same though, in any case I cannot reproduce either issue now.

**Comment by juliushaertl** [comment]
Board details was a different fix as well :) #2964
